pkg/store/fsm.go

```diff
@@ -383,7 +394,8 @@ func (s *Store) Snapshot() (raft.FSMSnapshot, error) {
 func (s *Store) Restore(closer io.ReadCloser) error {
 	var err error
 	var data persistData
-	err = json.NewDecoder(closer).Decode(&data)
+	snap, err := ioutil.ReadAll(closer)
+	err = json.Unmarshal(snap[8:], &data)
```

Comment:
You need to check the snap header, if the header equals is protocol version flag, you can do this.